### DISCUSSION TITLE
Is there a way to avoid copying artefacts from step to step

### DISCUSSION THREAD

**Discussion by suzizecat** [siliconcompiler/siliconcompiler#873] [Help (Q&A)] [answered]
Hi there !

Does SiliconCompiler have a setting allowing to *not* copy artefacts from one flow step to the other ?
Let say I have a huge design,  some of the beast I work on can have several GB of sources and synthesis outputs can weight around 20+ GB (Through industry tools). Therefore, copying all this stuff for each step would definitely takes a good amount of time and a very big amount of disk space.

Is there a way to work around this ?

**Comment by nmoroze** [comment, marked as answer]
Hi,

Good question! The idea of writing and reading artifacts in between build steps is pretty critical to SC's design, so there's no way to turn it off. However, it is worth pointing out that we actually use hard links to "copy" files from the outputs/ of one step to the inputs/ of another, so there's no actual duplication of data on disk. 

While our open-source reference scripts use tool-agnostic ASCII formats to communicate between steps, some tools support custom binary formats that may be smaller, so one could write reference scripts that use those (or perhaps perform compression in the tool pre_process() or post_process() functions) to make a slightly more disk space efficient flow.

Hope that helps!

**Reply by suzizecat** [reply]
I haven't seen that It was links rather than a copy, which makes perfectly sense !

Thanks for the answer !
